account_invoice_pricelist/models/account_move.py

```diff
@@ -0,0 +1,228 @@
+        self.filtered(lambda r: r.state == "draft").with_context(
+            check_move_validity=False
+        )._recompute_tax_lines()
+
+    def _reverse_move_vals(self, default_values, cancel=True):
+        move_vals = super(AccountMove, self)._reverse_move_vals(
+            default_values, cancel=cancel
+        )
+        if self.pricelist_id:
+            move_vals["pricelist_id"] = self.pricelist_id.id
+        return move_vals
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.onchange("product_id", "quantity")
+    def _onchange_product_id_account_invoice_pricelist(self):
+        for sel in self:
+            if not sel.move_id.pricelist_id:
+                return
+            sel.with_context(check_move_validity=False).update(
+                {"price_unit": sel._get_price_with_pricelist()}
+            )
+
+    @api.onchange("product_uom_id")
+    def _onchange_uom_id(self):
+        for sel in self:
+            if (
+                sel.move_id.is_invoice()
+                and sel.move_id.state == "draft"
+                and sel.move_id.pricelist_id
+            ):
+                price_unit = sel._get_computed_price_unit()
+                taxes = sel._get_computed_taxes()
+                if taxes and sel.move_id.fiscal_position_id:
+                    price_subtotal = sel._get_price_total_and_subtotal(
+                        price_unit=price_unit, taxes=taxes
+                    )["price_subtotal"]
+                    accounting_vals = sel._get_fields_onchange_subtotal(
+                        price_subtotal=price_subtotal,
+                        currency=self.move_id.company_currency_id,
+                    )
+                    amount_currency = accounting_vals["amount_currency"]
+                    price_unit = sel._get_fields_onchange_balance(
+                        amount_currency=amount_currency
+                    ).get("price_unit", price_unit)
+                sel.with_context(check_move_validity=False).update(
+                    {"price_unit": price_unit}
+                )
+            else:
+                super(AccountMoveLine, self)._onchange_uom_id()
+
+    def _get_real_price_currency(self, product, rule_id, qty, uom, pricelist_id):
+        PricelistItem = self.env["product.pricelist.item"]
+        field_name = "lst_price"
+        currency_id = None
+        product_currency = product.currency_id
+        if rule_id:
+            pricelist_item = PricelistItem.browse(rule_id)
+            while (
+                pricelist_item.base == "pricelist"
+                and pricelist_item.base_pricelist_id
+                and pricelist_item.base_pricelist_id.discount_policy
+                == "without_discount"
+            ):
+                price, rule_id = pricelist_item.base_pricelist_id.with_context(
+                    uom=uom.id
+                ).get_product_price_rule(product, qty, self.order_id.partner_id)
```

Comment:
where/what is this order_id field?

This seems basically a cut and paste of the sale order line code. Anyway it should be move_id.